codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -239,6 +239,7 @@ void removeStream(DefaultStream stream) {
         private State state = IDLE;
         private short weight = DEFAULT_PRIORITY_WEIGHT;
         private DefaultStream parent;
+        // TODO(nathanmittler): We no longer need a map. LinkedList?
```

Comment:
Iterative list type operations for "remove" may not be so bad if we anticipate the number of children to be small but every stream starts as dependent of the `connection` stream.  I think that means we are more likely to encounter a situation where a `remove` operation is invoked when the number of children is "large"?  Seems like the map will give us good performance for all use cases (inserts are a bit more expensive...and maybe more memory is needed), and the list may give us better performance for *most*/*some* use cases...but there are some where performance may be bad?